docs/modules/ROOT/pages/migration.adoc

```diff
@@ -211,7 +210,200 @@ yarl                1.9.4
 +
 [source,python]
 ----
-python3 migration.py
+python3 langchain-migration.py
 ----
 
 ...and you should see the same output as before, with no changes to your code required!
+
+== Example LlamaIndex migration
+
+Here is an application that uses LlamaIndex to index a set of documents.
+
+.llama-migration.py
+[%collapsible%open]
+====
+[source,python]
+----
+import os
+from llama_index.llama_dataset import download_llama_dataset
+from llama_index.vector_stores import AstraDBVectorStore
+from llama_index import VectorStoreIndex, SimpleDirectoryReader, StorageContext
+
+# Download and load dataset
+dataset = download_llama_dataset("PaulGrahamEssayDataset", "./data")
+documents = SimpleDirectoryReader("./data/source_files").load_data()
+
+# Display basic information about the documents
+print(f"Total documents: {len(documents)}")
+first_doc = documents[0]
+print(f"First document, id: {first_doc.doc_id}")
+print(f"First document, hash: {first_doc.hash}")
+print(f"First document, text ({len(first_doc.text)} characters):\n{'=' * 20}\n{first_doc.text[:360]} ...")
+
+# Setup AstraDB Vector Store
+astra_db_store = AstraDBVectorStore(
+    token=os.getenv("ASTRA_DB_APPLICATION_TOKEN"),
+    api_endpoint=os.getenv("ASTRA_DB_API_ENDPOINT"),
+    collection_name="test",
+    embedding_dimension=1536
+)
+
+# Create Storage Context and Index
+storage_context = StorageContext.from_defaults(vector_store=astra_db_store)
+index = VectorStoreIndex.from_documents(documents, storage_context=storage_context)
+
+# Query the index
+def execute_query(query_string, mode="default", top_k=3, mmr_prefetch_factor=None):
+    retriever = index.as_retriever(
+        vector_store_query_mode=mode,
+        similarity_top_k=top_k,
+        vector_store_kwargs={"mmr_prefetch_factor": mmr_prefetch_factor} if mmr_prefetch_factor else {}
+    )
+
+    nodes_with_scores = retriever.retrieve(query_string)
+
+    print(query_string)
+    print(f"Found {len(nodes_with_scores)} nodes.")
+    for idx, node_with_score in enumerate(nodes_with_scores):
+        print(f"    [{idx}] score = {node_with_score.score}")
+        print(f"        id    = {node_with_score.node.node_id}")
+        print(f"        text  = {node_with_score.node.text[:90]} ...")
+
+# Execute queries
+query_string_1 = "Why did the author choose to work on AI?"
+execute_query(query_string_1)
+execute_query(query_string_1, mode="mmr", mmr_prefetch_factor=4)
+----
+====
+
+. This application requires installation of the following packages:
++
+[source,python]
+----
+pip install llama-index
+----
++
+. Your application is tested and working at `llama-index` version `0.9.29`. But then, LlamaIndex version `0.10.1` changes the module to split every integration into its own PyPi package. Oh no, your application no longer works!
```

Comment:
to be fair here, users would have stick their dependency to 0.9.x so they wouldn't be affected by this change implicitly.

Also ragstack now can select specific integrations to be included in the stack but I believe we won't add anything by default except for astra (and maybe openai)